core/docker_service.go

```diff
@@ -261,6 +261,16 @@ func NewDockerService(
 	// Register prometheus metrics.
 	metrics.Register()
 
+	// Because we could not prevent users from stopping/deleting the pod containers outside CRI operations
+	// some stale entries may be left in ContainerStatsCache
+	// Do housekeeping every minute to clean up cache entries that are longer than 1 hour.
+	go func() {
+		for {
+			ContainerStatsCache.ClearByAge(time.Hour)
```

Comment:
How expensive can this call be? If it's not expensive, the following feels more correct to me:
```go
for range time.Tick(time.Minute) {
    ContainerStatsCache.ClearByAge(time.Hour)
}
```

Reply:
Not expensive, just iterating a map with at most hundreds of pointers. 
Updated per your suggestion in the new commit.